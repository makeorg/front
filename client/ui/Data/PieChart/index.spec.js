import React from 'react';
import renderer from 'react-test-renderer';
import { MockedPieChartResults } from 'Shared/types/__fixtures__/results.fixture';
import { PieChart } from './index';

jest.mock('./Styled', () => ({
  PieChartWrapperStyle: 'PieChartWrapperStyle',
  PieChartTitleStyle: 'PieChartTitleStyle',
  PieChartLegendStyle: 'PieChartLegendStyle',
  PieChartCanvasStyle: 'PieChartCanvasStyle',
}));

jest.mock('Client/ui/Elements/HiddenElements', () => ({
  HiddenItemStyle: 'HiddenItemStyle',
}));

describe('PieChart', () => {
  it('snapshot by default', () => {
    const component = renderer
      .create(
        <PieChart
          name={MockedPieChartResults.name}
          unit={MockedPieChartResults.unit}
          legend={MockedPieChartResults.legend}
          data={MockedPieChartResults.data}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
