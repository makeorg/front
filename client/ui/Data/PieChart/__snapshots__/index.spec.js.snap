// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieChart snapshot by default 1`] = `
<PieChartWrapperStyle>
  <PieChartTitleStyle>
    Pie Chart n° 1
  </PieChartTitleStyle>
  <HiddenItemStyle>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam.
    </p>
    <table>
      <caption>
        Pie Chart n° 1
      </caption>
      <thead>
        <tr>
          <th>
            consultation.results.table.name
          </th>
          <th>
            consultation.results.table.value_with_unit
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Justice sociale
          </td>
          <td>
            undefined%
          </td>
        </tr>
        <tr>
          <td>
            RSE et politique
          </td>
          <td>
            undefined%
          </td>
        </tr>
        <tr>
          <td>
            Conditions
          </td>
          <td>
            undefined%
          </td>
        </tr>
        <tr>
          <td>
            Intêret général
          </td>
          <td>
            undefined%
          </td>
        </tr>
        <tr>
          <td>
            Méthodes
          </td>
          <td>
            undefined%
          </td>
        </tr>
        <tr>
          <td>
            Emploi
          </td>
          <td>
            undefined%
          </td>
        </tr>
        <tr>
          <td>
            Croissance
          </td>
          <td>
            undefined%
          </td>
        </tr>
        <tr>
          <td>
            Code du
          </td>
          <td>
            undefined%
          </td>
        </tr>
        <tr>
          <td>
            Autre
          </td>
          <td>
            undefined%
          </td>
        </tr>
      </tbody>
    </table>
  </HiddenItemStyle>
  <PieChartCanvasStyle
    aria-hidden={true}
  />
  <PieChartLegendStyle
    aria-hidden={true}
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam.
  </PieChartLegendStyle>
</PieChartWrapperStyle>
`;
